salt/utils/decorators/__init__.py

```diff
@@ -179,17 +184,31 @@ def enforce_dependencies(cls, functions, kind, tgt_mod):
                     continue
 
                 # check if you have the dependency
-                elif dependency in frame.f_globals \
-                        or dependency in frame.f_locals:
-                    log.trace(
-                        'Dependency (%s) already loaded inside %s, skipping',
-                        dependency, mod_name
-                    )
-                    continue
+                elif dep_found:
+                    if 'version' in params:
+                        if salt.utils.versions.version_cmp(
+                                    dep_version,
+                                    params['version']
+                                ) >= 0:
+                            log.trace(
+                                'Dependency (%s) already loaded inside %s with '
+                                'version (%s), required (%s), skipping',
+                                dependency, mod_name, dep_version, params['version']
+                            )
+                            continue
+                    else:
+                        log.trace(
+                            'Dependency (%s) already loaded inside %s, skipping',
+                            dependency, mod_name
+                        )
+                        continue
 
                 log.trace(
-                    'Unloading %s.%s because dependency (%s) is not met',
-                    mod_name, func_name, dependency
+                    'Unloading %s.%s because dependency (%s%s) is not met',
```

Comment:
Was this missing a `.`?
```suggestion
                    'Unloading %s.%s because dependency (%s.%s) is not met',
```

Reply:
Nope, the 4th `%s` placeholder is filled with https://github.com/saltstack/salt/pull/55590/files#diff-7bd4f0af6512326b8c5a14e4ecb70b40R211 which starts with a space. Alternatively, I could move the space to this line (207) if you prefer.